### ISSUE TITLE
solo-python is broken

### ISSUE THREAD

**Issue by d4g** [NixOS/nixpkgs#78032]
**Describe the bug**
The solo-python module/script is broken.
The function "make-credential" fails.

**To Reproduce**
```
solo key make-credential                                                                     ~
Touch your authenticator to generate a credential...
Traceback (most recent call last):
  File "/nix/store/wdhccfihy44jcy7cac9ix8ll5spfq3vh-python3.7-solo-python-0.0.21/bin/.solo-wrapped", line 9, in <module>
    sys.exit(solo_cli())
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nix/store/zzmkaz2y9d9lzhkgimbfp6y20h8i3iv2-python3.7-click-7.0/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/wdhccfihy44jcy7cac9ix8ll5spfq3vh-python3.7-solo-python-0.0.21/lib/python3.7/site-packages/solo/cli/key.py", line 135, in make_credential
    host=host, user_id=user, serial=serial, output=True, prompt=prompt, udp=udp
  File "/nix/store/wdhccfihy44jcy7cac9ix8ll5spfq3vh-python3.7-solo-python-0.0.21/lib/python3.7/site-packages/solo/hmac_secret.py", line 44, in make_credential
    rp, user, challenge, extensions=hmac_ext.create_dict(), pin=pin
TypeError: make_credential() takes 2 positional arguments but 4 were given
```

Same behavior for python38

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.12, NixOS, 20.03pre209690.c438ce12a85 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-20.03pre209690.c438ce12a85"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
  - python37Packages.solo-python
  - python38Packages.solo-python
```

**Comment by d4g** [comment]
@wucke13

**Comment by veprbl** [comment]
The upstream issue solokeys/solo-python#52

**Comment by veprbl** [comment]
Needs a bump to 0.0.23

**Comment by d4g** [comment]
> The upstream issue [solokeys/solo-python#52](https://github.com/solokeys/solo-python/issues/52)

Thanks for that. I was expecting it to be a nixpkgs issue, because of the recent update, but I did not check for an upstream issue.

**Comment by wucke13** [comment]
I'll prepare a PR

**Comment by marsam** [comment]
Addressed by #78097
